LogsHandler/src/test/java/bootTest/IntegrationTest.java

```diff
@@ -42,32 +44,45 @@ public static void beforeAllTests() {
     public void testEndToEnd() {
         String key = RandomStringUtils.random(15, false, true);
         String jsonObjectAsString = "{\"message\":\"" + key + "\"}";
+        String account = "{'accountName':'fadi'}";
 
-        Response postResponse = webTarget.path("/index")
+        //create account
+        Response postResponse = webTarget.path("/create-account")
                 .request(MediaType.APPLICATION_JSON)
                 .header(HttpHeaders.USER_AGENT, userAgent)
-                .post(Entity.json(jsonObjectAsString));
-        assertEquals(200, postResponse.getStatus());
+                .header(HttpHeaders.CONTENT_TYPE, "application/json")
+                .post(Entity.json(account));
+        assertEquals(HttpURLConnection.HTTP_CREATED, postResponse.getStatus());
 
-        String result = postResponse.toString() + "\n" + postResponse.readEntity(String.class);
-        logger.debug(result);
+        String accountJsonString = postResponse.readEntity(String.class);
+        JsonObject accountJson = InfraUtil.stringToJson(accountJsonString);
+        String esIndexName = accountJson.get("esIndexName").getAsString();
+        String token = accountJson.get("token").getAsString();
 
+        //index
+        await().atMost(Duration.ofSeconds(7)).until(() -> {
+            Response indexResponse = webTarget.path("/index")
+                    .request(MediaType.APPLICATION_JSON)
+                    .header(HttpHeaders.USER_AGENT, userAgent)
+                    .header("X-ACCOUNT-TOKEN", token)
+                    .post(Entity.json(jsonObjectAsString));
+            assertNotNull(indexResponse);
+            return (200 == indexResponse.getStatus());
+        });
 
+        //search
         String header = "Macintosh";
         await().atMost(Duration.ofSeconds(7)).until(() -> {
-            Response searchResponse1 = webTarget.path("search")
-                    .queryParam("message", key)
-                    .queryParam("header", header)
+            Response searchResponse = webTarget.path("search")
                     .request(MediaType.APPLICATION_JSON)
+                    .header("X-ACCOUNT-TOKEN", token)
                     .get();
-            assertNotNull(searchResponse1);
-            String entity1 = searchResponse1.readEntity(String.class);
-            boolean isMessageIndexed1 = searchResponse1.getStatus() == HttpURLConnection.HTTP_OK;
+            assertNotNull(searchResponse);
+            String entity1 = searchResponse.readEntity(String.class);
+            boolean isMessageIndexed1 = searchResponse.getStatus() == HttpURLConnection.HTTP_OK;
             boolean isMessageFound1 = entity1.indexOf(key) > -1;
-
             return isMessageIndexed1 && isMessageFound1;
         });
-
     }
 
     @After
```

Comment:
Let's add some tests:
- Do /index with an invalid token - verify responded with 401
- Do /search with an invalid token - verify responded with 401
- Create two accounts, each one is sending logs, verify each one can see his logs ONLY